agent/api/task.go

```diff
@@ -865,22 +865,25 @@ func (task *Task) UpdateStatus() bool {
 
 // UpdateDesiredStatus sets the known status of the task
 func (task *Task) UpdateDesiredStatus() {
+	task.lock.Lock()
+	defer task.lock.Unlock()
 	task.updateTaskDesiredStatus()
 	task.updateContainerDesiredStatus()
+	task.updateResourceDesiredStatus()
 }
 
 // updateTaskDesiredStatus determines what status the task should properly be at based on the containers' statuses
 // Invariant: task desired status must be stopped if any essential container is stopped
 func (task *Task) updateTaskDesiredStatus() {
```

Comment:
Please also update the method name to indicate it's not protected by a lock. Same for the others.